src/metabase/lib/js.cljs

```diff
@@ -445,13 +450,17 @@
   (lib.core/remove-field a-query stage-number column))
 
 (defn ^:export find-visible-column-for-legacy-ref
-  "Return the visible column in `a-query` at `stage-number` referenced by `legacy-ref`."
+  "Like [[find-visible-column-for-ref]], but takes a legacy MBQL reference instead of a pMBQL one. This is currently
+  only meant for use with `:field` clauses."
   [a-query stage-number legacy-ref]
-  (let [ref (-> legacy-ref
-                (js->clj :keywordize-keys true)
-                (update 0 keyword)
-                convert/->pMBQL)]
-    (lib.core/find-visible-column-for-ref a-query stage-number ref)))
+  ;; [[lib.convert/legacy-ref->pMBQL]] will handle JS -> Clj conversion as needed
+  (lib.core/find-visible-column-for-legacy-ref a-query stage-number legacy-ref))
```

Comment:
I dumbed down some of the implementations in `lib.js`, since the new helper function takes care of this stuff, and moved JS-specific logic into the actual implementations where appropriate and added tests for things.